src/main/java/swarm/robot/sensors/CompassSensor.java

```diff
@@ -21,7 +21,11 @@ private enum mqttTopic {
     }
 
     private HashMap<CompassSensor.mqttTopic, String> topicsSub = new HashMap<>();
-    private double heading;
+    private double reading;
+
+//    private static final double TOLERANCE = 0.000001;  // Value should be closer to zero
+
+    private static final double NORTH = 0.0;  // Should measure relative to a reference (here, to the horizontal)
```

Comment:
@dhmudalige shouldn't this value be 90.0?

Reply:
> @dhmudalige shouldn't this value be 90.0?

Here, I have just assumed it as 0. Actual value should be measured using a compass with reference to the horizontal (counter-clockwise => +)

Reply:
![image](https://github.com/Pera-Swarm/robot-library-java/assets/11540782/69f89d97-8ecc-4c5d-aa00-ed544a74cddb)

I thought you were using this coordination system. (North is the top side of the testbed)

Reply:
>**North is the top side of the testbed**
If this is so, you are right.